UndertaleModTool/Scripts/Community Scripts/TouchControlsEnabler.csx

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Linq;
+using UndertaleModLib.Util;
+
+EnsureDataLoaded();
+
+string display_name = Data?.GeneralInfo?.DisplayName?.Content.ToLower();
+
+if (display_name != "deltarune chapter 1 & 2" && display_name != "deltarune chapter 1&2" && display_name != "undertale")
+{
+    ScriptError("Error 0: Only compatible with Undertale & Deltarune");
+    return;
+}
+
+string dataPath = Path.Combine(Path.GetDirectoryName(ScriptPath), "TouchControls_data");
+
+int lastTextPage = Data.EmbeddedTextures.Count - 1;
+int lastTextPageItem = Data.TexturePageItems.Count - 1;
+
+Dictionary<string, UndertaleEmbeddedTexture> textures = new Dictionary<string, UndertaleEmbeddedTexture>();
+
+UndertaleEmbeddedTexture controlsTexturePage = new UndertaleEmbeddedTexture()
+{
+    Name = new UndertaleString("Texture " + Data.EmbeddedTextures.Count),
+};
+controlsTexturePage.TextureData.TextureBlob = File.ReadAllBytes(Path.Combine(dataPath, "controls.png"));
+Data.EmbeddedTextures.Add(controlsTexturePage);
+textures.Add(Path.GetFileName(Path.Combine(dataPath, "controls.png")), controlsTexturePage);
+
+UndertaleEmbeddedTexture black = new UndertaleEmbeddedTexture()
+{
+    Name = new UndertaleString("Texture " + Data.EmbeddedTextures.Count),
+};
+black.TextureData.TextureBlob = File.ReadAllBytes(Path.Combine(dataPath, "black.png"));
+Data.EmbeddedTextures.Add(black);
+textures.Add(Path.GetFileName(Path.Combine(dataPath, "black.png")), black);
+
+UndertaleTexturePageItem AddNewTexturePageItem(ushort sourceX, ushort sourceY, ushort sourceWidth, ushort sourceHeight, string texturePageName)
+{
+    var name = new UndertaleString("PageItem " + Data.TexturePageItems.Count);
```

Comment:
giving names to texture page items is technically unnecessary. see texture page comment.

Reply:
Okie.